src/conn/conn_dhandle.c

```diff
@@ -730,7 +732,10 @@ __wt_conn_dhandle_discard_single(WT_SESSION_IMPL *session, int final)
 
 	if (F_ISSET(dhandle, WT_DHANDLE_OPEN)) {
 		ret = __wt_conn_btree_sync_and_close(session, 0);
-		if (!final)
+		if (final && ret != 0)
+			__wt_err(session, ret,
+			    "Final close of %s failed", dhandle->name);
```

Comment:
I'm open to suggestions about what to do here: previously, we silently ignored failures...